### PR TITLE
Verify datamodel name on flow init

### PR DESCRIPTION
**Pull request by DatGuyJonathan**
Screenshot 2024-05-06 at 1 47 42 PM

### CODE REVIEW

File: apps/framework-cli/src/cli/routines/flow.rs [hunk 1]
```diff
@@ -76,12 +76,75 @@ pub fn create_flow_file(
...
+pub fn verify_datamodel(project: &Project, datamodel: String) -> anyhow::Result<()> {
+    let child = std::process::Command::new("grep")
+        .arg("-r")
+        .arg("--include=*.ts")
+        .arg("-E")
+        .arg(format!("export\\s+interface\\s+\\b{}\\b", &datamodel))
+        .arg(project.schemas_dir())
+        .stdout(Stdio::piped())
+        .stderr(Stdio::piped())
+        .spawn()?;
```
Comment (DatGuyJonathan) [inline apps/framework-cli/src/cli/routines/flow.rs:134]: One option is to crawl the schema, get framework objects, and check against that. It seems heavy for just an init command.

Another option is to check the .moose folder, but that requires dev to have been run.

Grep seems lightweight enough and works on Mac/Linux. LMK what y'all think.

Comment (callicles) [reply apps/framework-cli/src/cli/routines/flow.rs:134]: That's a creative solution!

Quick question - should we fail if the datamodel doesn't exist?

Comment (DatGuyJonathan) [reply apps/framework-cli/src/cli/routines/flow.rs:134]: Their order of operation could vary. They could create their data models first, then the flow, in which case we wouldn't print next steps. They could create the flow file first, then their data models, in which case we're just reminding them to add their data models by printing these next steps. I think that's a valid workflow. Feels like the reminder is gentler & they won't have to come back & re-run the flow init command.

Comment (callicles) [reply apps/framework-cli/src/cli/routines/flow.rs:134]: 🤔  that feels fair for the output but does it work for the input? ie can you write a flow without having defined the data you will take as input?

Comment (callicles) [reply apps/framework-cli/src/cli/routines/flow.rs:134]: Your other PR should cover this

**Review by callicles — APPROVED** [review]

**Pull request merged**